com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs

```diff
@@ -422,6 +422,16 @@ bool SaveAsImplementation()
 
                 if (newPath != path)
                 {
+                    var subGraphNodes = graphObject.graph.GetNodes<SubGraphNode>();
```

Comment:
I'm concerned that this only catches a single level of recursion. Please see the recursion check in `SubGraphNode.ValidateNode()` for a deeper way to do it. And if that code can be made a static function on SubGraphNode that can be called there and here, even better.

Reply:
Fixed to use the pre-baked descendents list, which can catch any depth of recursion.